lib/uwb/include/uwb/UwbSession.hxx

```diff
@@ -118,14 +92,39 @@ public:
     SetSessionStatus(const uwb::protocol::fira::UwbSessionStatus& status);
 
     /**
-     * @brief Temporarily public function to directly add a peer to m_peers
+     * @brief Temporarily public function to directly add a list of peers to m_peers
      *
-     * @param peerAddress
+     * @param peers
      */
+    template <class View>
+    // clang-format off
+    requires std::ranges::view<View> &&
+    std::same_as<std::decay_t<std::ranges::range_reference_t<View>>, uwb::protocol::fira::UwbMulticastListStatus>
+    // clang-format on
     void
-    InsertPeer(const uwb::UwbMacAddress& peerAddress);
+    InsertPeers(View peers)
+    {
+        std::vector<UwbPeer> uwbPeers;
+        {
+            std::scoped_lock peersLock{ m_peerGate };
+            for (const auto& peer : peers) {
+                InsertPeerImpl(peer.ControleeMacAddress);
+                uwbPeers.push_back(UwbPeer{ peer.ControleeMacAddress });
+            }
+        }
+        auto callbacks = m_callbacks.lock();
+        callbacks->OnSessionMembershipChanged(this, uwbPeers, std::vector<UwbPeer>{});
```

Comment:
nit: this can just be empty braces
```suggestion
        auto callbacks = m_callbacks.lock();
        callbacks->OnSessionMembershipChanged(this, uwbPeers, {});
```